core/components/subscribe/_actions/subscribe.ts

```diff
@@ -0,0 +1,25 @@
+'use server';
+
+import { SubmissionResult } from '@conform-to/react';
+import { parseWithZod } from '@conform-to/zod';
+import { getTranslations } from 'next-intl/server';
+
+import { schema } from '@/vibes/soul/primitives/inline-email-form/schema';
+
+export const subscribe = async (
+  _lastResult: { lastResult: SubmissionResult | null },
+  formData: FormData,
+) => {
+  const t = await getTranslations('Components.Subscribe');
+
+  const submission = parseWithZod(formData, { schema });
+
+  if (submission.status !== 'success') {
+    return { lastResult: submission.reply() };
```

Comment:
Should we reply with a form error here? This will show, for example, if the user provides an invalid email address

Reply:
This is what I don't get, it already shows this error by just replying.

Reply:
Ah yes you're correct. We only need to pass a form error if we handle an error with try/catch and we want to send an error message that wasn't from an error as a result of parsing. So there's no need to do so in this case. My bad